pandas/plotting/_matplotlib/core.py

```diff
@@ -965,7 +967,10 @@ def _make_plot(self):
         elif color is not None:
             c_values = color
         elif c_is_column:
-            c_values = self.data[c].values
+            if not is_categorical_dtype(self.data[c]):
```

Comment:
nit but instead of `not is_categorical_dtype` can you leave this as `is_categorcial_dtype` and switch the branches?